core/src/main/scala/org/elasticmq/actor/QueueManagerActor.scala

```diff
@@ -23,9 +24,19 @@ class QueueManagerActor(nowProvider: NowProvider) extends ReplyingActor with Log
           Left(new QueueAlreadyExists(queueData.name))
         } else {
           logger.info(s"Creating queue $queueData")
-          val actor = createQueueActor(nowProvider, queueData)
-          queues(queueData.name) = actor
-          Right(actor)
+          for {
```

Comment:
[minor] I think this would be more readable by explicitly handling left/rigth:

```
Limits.verifyQueueName(...) match {
  case Left(msg) => Left(QCE(...))
  case Right(_) => val actor = ...
}
```